### PR TITLE
Remove some unneeded checks

### PR DESCRIPTION
**Pull request by DanielRuf**
**Type:**
bug

The following has been addressed in the PR:

*  There are some unnecessary checks.

**Description:**

Remove some unnecessary checks.

### CODE REVIEW

File: src/lib/up-storage.js [hunk 1]
```diff
@@ -540,7 +540,7 @@ class ProxyStorage implements IProxy {
       //
       if (this.proxy === false) {
         headers['X-Forwarded-For'] = (
-          req && req.headers['x-forwarded-for']
+          req.headers['x-forwarded-for']
```
Comment (juanpicado) [inline src/lib/up-storage.js]: Actually can be undefined 

```
// line 447
const readStream = this.request({
      uri_full: url,
      encoding: null,
      headers: {
        Accept: contentTypeAccept,
      },
    });
```

Comment (DanielRuf) [reply src/lib/up-storage.js]: See https://github.com/DanielRuf/verdaccio/blob/46787f3306da87684ab17d2452318c0108d14010/src/lib/up-storage.js#L534 which already checks it or did I oversee something?

Comment (juanpicado) [reply src/lib/up-storage.js]: I did not see that 👍  My bad.

Comment (juanpicado) [reply src/lib/up-storage.js]: Thanks @DanielRuf ! You are on 🔥 today.

**Pull request merged**